packages/loader/index.js

```diff
@@ -1,27 +1,39 @@
 const {getOptions} = require('loader-utils')
-const mdx = require('@mdx-js/mdx')
+const {createCompiler} = require('@mdx-js/mdx')
 
 const DEFAULT_RENDERER = `
 import React from 'react'
 import {mdx} from '@mdx-js/react'
 `
 
+const pragma = `
+/* @jsxRuntime classic */
+/* @jsx mdx */
+/* @jsxFrag mdx.Fragment */
+`
+
+const compilerCache = new Map()
```

Comment:
Could cache size be a concern?
Would FIFO cache or LRU cache make sense here?
Maybe something like https://www.npmjs.com/package/quick-lru?

Reply:
The maximum cache size is the times `@mdx-js/loader` is used across all Webpack configurations of a project. Typically this is 1. In exceptional situations there may be more. I don’t think this will grow very big.

I also tried using `WeakMap`, but that doesn’t support primitive keys, which `query` can be.